root/admin/attributes/types.js

```diff
@@ -9,54 +9,62 @@
 
 export type CreateOrEditAttributePropsT =
   | $ReadOnly<{
-    +action: 'add' | 'edit',
-    +form: AttributeEditGenericFormT,
-    +parentSelectOptions: SelectOptionsT,
-    +type:
-      | 'AreaType'
-      | 'ArtistType'
-      | 'CoverArtType'
-      | 'EventType'
-      | 'Gender'
-      | 'InstrumentType'
-      | 'LabelType'
-      | 'PlaceType'
-      | 'ReleaseGroupType'
-      | 'ReleaseGroupSecondaryType'
-      | 'ReleaseStatus'
-      | 'ReleasePackaging'
-      | 'WorkType',
-  }>
+      +action: 'add' | 'edit',
+      +form: AttributeEditGenericFormT,
+      +parentSelectOptions: SelectOptionsT,
+      +type:
+        | 'AreaType'
+        | 'ArtistType'
+        | 'CoverArtType'
+        | 'EventType'
+        | 'Gender'
+        | 'InstrumentType'
+        | 'LabelType'
+        | 'PlaceType'
+        | 'ReleaseGroupType'
+        | 'ReleaseGroupSecondaryType'
+        | 'ReleaseStatus'
+        | 'ReleasePackaging'
+        | 'WorkType',
+    }>
   | $ReadOnly<{
-    +action: 'add' | 'edit',
-    +entityTypeSelectOptions:
-      {[entityType: CoreEntityTypeT]: CoreEntityTypeT},
-    +form: AttributeEditFormWithEntityTypeT,
-    +parentSelectOptions: SelectOptionsT,
-    +type: 'CollectionType' | 'SeriesType',
-  }>
+      +action: 'add' | 'edit',
+      +entityTypeSelectOptions:
+        {[entityType: CoreEntityTypeT]: CoreEntityTypeT},
+      +form: AttributeEditFormWithEntityTypeT,
+      +parentSelectOptions: SelectOptionsT,
+      +type: 'CollectionType',
+    }>
   | $ReadOnly<{
-    +action: 'add' | 'edit',
-    +form: LanguageEditFormT,
-    +type: 'Language',
-  }>
+      +action: 'add' | 'edit',
+      +entityTypeSelectOptions:
+        {[entityType: CoreEntityTypeT]: CoreEntityTypeT},
+      +form: AttributeEditFormWithEntityTypeT,
+      +parentSelectOptions: SelectOptionsT,
+      +type: 'SeriesType',
+    }>
```

Comment:
Huh, so even if they are the same, Flow doesn't allow using a union type?

Reply:
Yes, not sure why, but it broke Flow's inference for some reason.